src/ng/cacheFactory.js

```diff
@@ -223,6 +223,8 @@ function $CacheFactoryProvider() {
             delete lruHash[key];
           }
 
+          if (!(key in data)) return;
```

Comment:
This will still has unexpected results if `key` is prototypally inherited by `data`.
You could use `!data.hasOwnProperty(key)`. It might also be possible to use `data = createMap()` and not have to deal with the prototype chain (but I haven't checked if more modifications are needed in other places).